classes/widgets/class-badge-content.php

```diff
@@ -21,6 +21,22 @@ final class Badge_Content extends Widget {
 	 */
 	protected $id = 'badge-content';
 
+	/**
+	 * Whether we should render the widget or not.
+	 *
+	 * @return bool
+	 */
+	protected function should_render() {
+		$details = $this->get_badge_details();
+		if (
+			'awesome-author' === $details['badge']['id']
```

Comment:
Does it matter which badge it is? If the progress is 100, shouldn't we never render it here?

Reply:
If it's 100, then the badge is not pending so internally we move on to the next one.
The only scenario where we could have displayed `100` is if it was the final badge. This is handled in the `get_badge_details()` method:
https://github.com/Emilia-Capital/progress-planner/blob/fdd04d3d7bf064fe7e3b8a8f6e311839622d4b47/classes/widgets/class-badge-content.php#L103-L109

Reply:
Hmmm I get what you mean now.
The only scenario where `100` would show, is if it was the final badge.

Makes sense, I'll go ahead and remove the check 👍